app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -161,4 +178,48 @@ private void addItemsOnSocialSitualtionSpinner(){
         socialSituationSpinner.setAdapter(socialSituationAdapter);
     }
 
+
+    /**
+     * Enables the My Location layer if the fine location permission has been granted.
+     */
+
+    private void enableMyLocation() {
+        if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION)
+                != PackageManager.PERMISSION_GRANTED) {
+            // Permission to access the location is missing.
+            PermissionUtils.requestPermission(this, LOCATION_PERMISSION_REQUEST_CODE,
+                    android.Manifest.permission.ACCESS_FINE_LOCATION, true);
+        }
+    }
+
+    private void toggleLocation(boolean isChecked, CompoundButton button){
```

Comment:
thanks. there will be a double longitude, double lattitude if location is on, or both going to be null if location is off